src/vm/customattribute.cpp

```diff
@@ -165,11 +165,11 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         ULONG length = pCaVal->arr.length;
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
-        if (length == (ULONG)-1)
```

Comment:
Amending my prior comment about "replicating the GC_PROTECT" - the correct change is to replace the existing GC_PROTECT with one that guards the whole function, starting from the line after "ZeroMemory(&gc, sizeof(gc)). Right now, the GC_PROTECT regions covers <b>just</b> enough to get by but that structure is an accident waiting to happen.